src/Alert.php

```diff
@@ -4,269 +4,358 @@
+     *
+     * @return self A new instance of the current class with the specified toggle attributes.
+     *
+     * @see {\Yiisoft\Html\Html::renderTagAttributes()} for details on how attributes are being rendered.
      */
-    public function info(): self
+    public function toggleAttributes(array $value): self
     {
-        return $this->addClassNames('alert-info');
+        $new = clone $this;
+        $new->toggleAttributes = $value;
+
+        return $new;
     }
 
     /**
-     * Short method for light alert type
+     * Sets the toggle component to be rendered as a link instead of a button.
+     *
+     * @return self A new instance of the current class with the toggle set as a link.
      */
-    public function light(): self
+    public function toggleLink(): self
     {
-        return $this->addClassNames('alert-light');
+        $new = clone $this;
+        $new->toggleLink = true;
+
+        return $new;
     }
 
     /**
-     * Short method for dark alert type
+     * Set the alert type. The following options are allowed:
+     * - `Type::PRIMARY`: Primary alert.
+     * - `Type::SECONDARY`: Secondary alert.
+     * - `Type::SUCCESS`: Success alert.
+     * - `Type::DANGER`: Danger alert.
+     * - `Type::WARNING`: Warning alert.
+     * - `Type::INFO`: Info alert.
+     * - `Type::LIGHT`: Light alert.
+     * - `Type::DARK`: Dark alert.
+     *
+     * @return self A new instance of the current class with the specified alert type.
      */
-    public function dark(): self
+    public function type(Type $type): self
     {
-        return $this->addClassNames('alert-dark');
+        $new = clone $this;
+        Html::addCssClass($new->attributes, ['persistent' => self::CLASS_COMPONENT, 'alert-' . $type->value]);
+
+        return $new;
     }
 
     /**
-     * Render header tag
+     * Run the alert widget.
+     *
+     * @return string The HTML representation of the element.
      */
-    private function renderHeader(): ?string
+    public function render(): string
     {
-        if ($this->header === null) {
-            return null;
+        $attributes = $this->attributes;
+        $attributes['role'] = 'alert';
+        $content = '';
+        $id = $this->id;
+        $toggle = '';
+
+        Html::addCssClass($attributes, ['persistent' => 'alert']);
+
+        if ($this->dismissing) {
+            $toggle = Toggle::widget()->attributes($this->toggleAttributes)->type(ToggleType::TYPE_DISMISING);
+
+            if ($this->toggleLink) {
+                $toggle = $toggle->link();
+            }
         }
 
-        $options = $this->headerOptions;
-        $encode = ArrayHelper::remove($options, 'encode', true);
+        $content = strtr(
+            $this->templateContent,
+            [
+                '{header}' => $this->renderHeader(),
+                '{body}' => $this->body,
+                '{toggle}' => $toggle,
+            ]
+        );
+
+        $content = preg_replace("/\n{2}/", "\n", $content);
```

Comment:
What is it for?

Reply:
To clear the content, when the widget generates two `\n\n` .